### PR TITLE
feat: elevate adapter common configuration

### PR DESCRIPTION
**Pull request by wangyizhi1**
#28 

/cc @OrangeBao 
/cc @xyz2277 
/cc @wuyingjun-lucky

### CODE REVIEW

File: pkg/knode-manager/adapters/k8s/pod.go [hunk 1]
```diff
@@ -65,57 +64,82 @@ type PodAdapter struct {
...
+		clientCache: clientCache{
+			podLister:    ac.PodInformer.Lister(),
+			nsLister:     ac.NamespaceInformer.Lister(),
+			cmLister:     ac.ConfigmapInformer.Lister(),
+			secretLister: ac.SecretInformer.Lister(),
+			nodeLister:   ac.NodeInformer.Lister(),
+		},
+		rm:         ac.ResourceManager,
+		updatedPod: make(chan *corev1.Pod, 100000),
+		stopCh:     ctx.Done(),
```
Comment (OrangeBao) [inline pkg/knode-manager/adapters/k8s/pod.go:87]: Suggest declaring the variable `updatePod` in  func `NewKnode`

Comment (wangyizhi1) [reply pkg/knode-manager/adapters/k8s/pod.go:87]: I recommend that the ``podAdapter`` takes care of its own required business logic going forward.  I have already moved the relevant event handlers from the ``nodeAdapter``.  Multiple adapters share the same ``indexer``, but their processing logic is independent.

Comment (OrangeBao) [reply pkg/knode-manager/adapters/k8s/pod.go:87]: /lgtm

Comment (OrangeBao) [reply pkg/knode-manager/adapters/k8s/pod.go:87]: /approve

File: pkg/knode-manager/adapters/k8s/pod.go [hunk 2]
```diff
@@ -65,57 +64,82 @@ type PodAdapter struct {
...
 	}
 
-	// master config, maybe a real node or a pod
-	master, err := utils.NewClient(cfg.ConfigPath, func(config *rest.Config) {
-		// config.QPS = float32(opts.KubeAPIQPS)
-		// config.Burst = int(opts.KubeAPIBurst)
+	_, err := ac.PodInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    adapter.addPod,
+		UpdateFunc: adapter.updatePod,
```
Comment (OrangeBao) [inline pkg/knode-manager/adapters/k8s/pod.go:92]: /approve

**Pull request merged**